lang/java/reef-common/src/main/java/org/apache/reef/driver/restart/DriverRestartManager.java

```diff
@@ -168,4 +172,69 @@ public synchronized void recordAllocatedEvaluator(final String id) {
   public synchronized void recordRemovedEvaluator(final String id) {
     driverRuntimeRestartManager.recordRemovedEvaluator(id);
   }
+
+  /**
+   * Signals to the {@link DriverRestartManager} that an evaluator is to be expected to report back after restart.
+   */
+  public synchronized void setEvaluatorExpected(final String evaluatorId) {
```

Comment:
I preferred keeping it public, since some of the other ``setEvaluator`` calls are needed to be called from the outside, and left this here to keep it consistent. What do you think?